src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -2149,6 +2159,111 @@ static void assertBindIdentifierTargets(InMethodBinder? inMethodBinder, Concurre
 #endif
         }
 
+        /// <summary>
+        /// Report a diagnostic for any 'field' or 'value' identifier in the bound tree where the
+        /// meaning will change when the identifier is considered a contextual keyword.
+        /// </summary>
+        private static void ReportFieldOrValueContextualKeywordConflicts(MethodSymbol method, BoundNode node, BindingDiagnosticBag diagnostics)
```

Comment:
I'm not sure I love this overall approach (additional scan through syntax and bound nodes). But I convinced me this is okay: 1. the performance impact seems cheap and the check is gated on older LangVer anyways, 2. we won't have to maintain the list below as new syntax nodes won't have to be handled (again because of LangVer).
Still, I'm curious, what made you prefer this approach over adding a check in `BindIdentifier` (conditional on being inside specific accessors)?

Reply:
> what made you prefer this approach over adding a check in `BindIdentifier`

I couldn't find a single method that would be called for each identifier. For instance, `BindIdentifier()` handles `SimpleNameSyntax` only, which covers `IdentifierNameSyntax` and `GenericNameSyntax` but not the others.

Reply:
Something bothers me about this approach in general and the place where the work is done. I cannot point to a specific problem or suggest a specific alternative at the moment. I do not mean to block the PR, but I am not comfortable signing off on it at the moment. Sorry.

Reply:
The current approach reports diagnostics after binding, but based on the syntax tree rather than the bound tree. (The bound tree is only used to determine whether `value` refers to the implicit accessor parameter.)

There are several alternative approaches that have been mentioned:

1. Determine the appropriate context in the *parser*, and mark identifiers that will become keywords in the *parser* (as @CyrusNajmabadi [suggested](https://github.com/dotnet/roslyn/pull/73570#discussion_r1614261276)), and either:
   1. Report diagnostics for `field` in the parser, but *do not* report diagnostics for `value` (as @RikkiGibson [suggested](https://github.com/dotnet/roslyn/pull/73570#discussion_r1614359697)), or
   1. Report diagnostics for `field` and `value` in *binding* as each identifier is bound.
1. Determine the appropriate context in *binding* (no changes to the parser), and report diagnostics for `field` and `value` in *binding* as each identifier is bound.

It looks like there are ~20 or so locations in binding where identifiers are bound. If we want to report diagnostics in binding, we'll probably need to call a helper method at each of those locations.

I think I'm comfortable with any of those three options.

Reply:
Implemented:
> 2. Determine the appropriate context in _binding_ (no changes to the parser), and report diagnostics for `field` and `value` in _binding_ as each identifier is bound.